lib/src/main/java/org/esbtools/eventhandler/NotificationRepository.java

```diff
@@ -21,16 +21,35 @@
 import java.util.Collection;
 import java.util.List;
 
+/**
+ * Abstracts a transactional backing store and staging area for {@link Notification notifications}.
+ *
+ * <p>A repository is responsible for handling CRUD and data parsing operations around these
+ * objects, in whatever scheme necessary.
+ *
+ * <p>Production implementations are expected to be thread safe, even across a network.
+ */
 public interface NotificationRepository {
 
     /**
-     * Retrieves the oldest {@code maxEvents} {@link Notification notifications}, oldest first.
+     * Retrieves the oldest {@code maxNotifications} {@link Notification notifications}, oldest
+     * first.
      *
      * <p>Once retrieved, a notification should not be retrieved again, atomically. That is, many
-     * threads looking up notifications at the same time should all et a unique non-overlapping sample
-     * of the oldest notifications. Subsequent calls should always return a unique set.
+     * threads looking up notifications at the same time should all et a unique non-overlapping
+     * sample of the oldest notifications. Subsequent calls should always return a unique set.
```

Comment:
Rate is num in batch * interval per batch * threads in cluster right? My thinking is just that larger batches are more efficient since lightblue and the event handler can do more work per network call. You can still scale horizontally as needed and get more total threads.

I'll go with per notification id then, but I'll put randomization for another PR since it will need to be weighted so highest priority notifications/events don't have chance of being ignored too long.